web/packages/teleterm/src/ui/Vnet/DocumentVnetDiagReport.tsx

```diff
@@ -0,0 +1,301 @@
+          >
+            <H1>VNet Diagnostic Report</H1>
+
+            <Flex gap={2}>
+              {/* TODO: Implement buttons. */}
+              <HoverTooltip tipContent="Run Diagnostics Again">
+                <Button intent="neutral" p={1}>
+                  <Refresh size="medium" />
+                </Button>
+              </HoverTooltip>
+
+              <HoverTooltip tipContent="Copy Report to Clipboard">
+                <Button intent="neutral" p={1}>
+                  <Copy size="medium" />
+                </Button>
+              </HoverTooltip>
+
+              <HoverTooltip tipContent="Save Report to File">
+                <Button intent="neutral" p={1}>
+                  <Download size="medium" />
+                </Button>
+              </HoverTooltip>
+            </Flex>
+          </Flex>
+
+          {networkStackAttempt.status === diag.CheckAttemptStatus.ERROR && (
+            <>
+              <P2>Created at: {createdAt}</P2>
+              <Alert
+                kind="danger"
+                details={<P2>{networkStackAttempt.error}</P2>}
+              >
+                Network details could not be determined.
+              </Alert>
+            </>
+          )}
+
+          {networkStackAttempt.status === diag.CheckAttemptStatus.OK && (
+            <P2>
+              Created at: {createdAt}
+              <br />
+              Network interface: <code>{networkStack.interfaceName}</code>
+              <br />
+              IPv4 CIDR {pluralize(networkStack.ipv4CidrRanges.length, 'range')}
+              : <code>{networkStack.ipv4CidrRanges.join(', ')}</code>
+              <br />
+              IPv6 prefix: <code>{networkStack.ipv6Prefix}</code>
+              <br />
+              DNS {pluralize(networkStack.dnsZones.length, 'zone')}:{' '}
+              <code>{networkStack.dnsZones.join(', ')}</code>
+            </P2>
+          )}
+        </Stack>
+
+        {report.checks.map(checkAttempt => (
+          <CheckAttempt
+            // tshd promises that checkAttempt.checkReport.report.oneofKind is
+            // 1) always present even if the check fails to complete
+            // 2) unique
+            key={checkAttempt.checkReport.report.oneofKind}
+            checkAttempt={checkAttempt}
+          />
+        ))}
+      </Stack>
+    </Document>
+  );
+}
+
+/**
+ * CheckAttempt displays the result of attempting to run an individual check along with the outputs
+ * of the accompanying commands. The commands are displayed even if the check itself failed to run.
+ */
+const CheckAttempt = ({
+  checkAttempt,
+}: {
+  checkAttempt: diag.CheckAttempt;
+}) => {
+  const reportOneof = checkAttempt.checkReport.report.oneofKind;
+  const displayDetails = reportOneofDisplayDetails[reportOneof];
+
+  return (
+    <Stack gap={2} width="100%">
+      {!displayDetails ? (
+        <Alert kind="danger">
+          Cannot display the result from an unsupported check {reportOneof}.
```

Comment:
Nit: I'm not sure if we have a rule for it, but I believe titles in our alerts usually don't end with a dot.
But if it was done on purpose, then I'd be consistent and add it in line 180. too.

Reply:
I'll remove the dots then.